File: content/bdd-with-cucumber/05/code/js/01-remove-triangulation-scenario/features/step_definitions/steps.js
const { Given, When, Then, Before } = require('@cucumber/cucumber')
const { assertThat, contains, not } = require('hamjest')

const { Person, Network } = require('../../src/shouty')

Before(function () {
  this.people = {}
})

Given('the range is {int}', function (range) {
  this.network = new Network(range)
})

Given('a person named {word} is located at {int}', function (name, location) {
  this.people[name] = new Person(this.network, location)
})

When('Sean shouts {string}', function (message) {
  this.people['Sean'].shout(message)
  this.messageFromSean = message
})

Then('Lucy should hear Sean\'s message', function () {
  assertThat(this.people['Lucy'].messagesHeard(), contains(this.messageFromSean))
})

Then('Larry should not hear Sean\'s message', function () {
  assertThat(this.people['Larry'].messagesHeard(), not(contains(this.messageFromSean)))
})

File: content/bdd-with-cucumber/05/code/ruby/01-remove-triangulation-scenario/features/step_definitions/steps.rb
require 'shouty'

Before do
  @people = {}
end

Given "the range is {int}" do |range|
  @network = Shouty::Network.new(range)
end

Given "a person named {word} is located at {int}" do |name, location|
  @people[name] = Shouty::Person.new(@network, location)
end

When "Sean shouts {string}" do |message|
  @people["Sean"].shout(message)
  @message_from_sean = message
end

Then "Lucy should hear Sean's message" do
  expect(@people['Lucy'].messages_heard).to eq [@message_from_sean]
end

Then "Larry should not hear Sean's message" do
  expect(@people['Larry'].messages_heard).not_to include(@message_from_sean)
end
